examples/nodes/torchata_nodes_basics.ipynb

```diff
@@ -0,0 +1,485 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0, 1, 2, 3]\n",
+      "[4, 5, 6, 7]\n",
+      "[8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batcher = Batcher(source, batch_size = 4, drop_last = False)\n",
+    "for batch in batcher:\n",
+    "    print(batch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e11543b9-c2dd-4861-af24-0eb3a1f3ddb9",
+   "metadata": {},
+   "source": [
+    "If we try to use this batcher over multiple epochs, we will need to reset it after every epoch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "d89e36a9-cc9c-4b55-acba-dbcaf8d53407",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch = 0  Batch = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n",
+      "Epoch = 1  Batch = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batcher = Batcher(source, batch_size = 10)\n",
+    "num_epochs = 2\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    for batch in batcher:\n",
+    "        print(f\"Epoch = {epoch}\", f\" Batch = {batch}\")\n",
+    "    batcher.reset()\n",
+    "    \n",
+    "# This is one extra step than traditional dataloader, we can actually wrap the batcher in a Loader to skip that\n",
+    "# Let's look at Loader in the next cell"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2232823-0927-432b-9171-fb2bf38a7205",
+   "metadata": {},
+   "source": [
+    "#### Loader\n",
+    "As you can see, we get a batch in every epoch, without even needing to reset the loader!!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "4f8a1830-49a2-4dfd-b17b-acaca7ac2e98",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch = 0  Batch = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n",
+      "Epoch = 1  Batch = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from torchdata.nodes import Loader\n",
+    "batcher = Batcher(source, batch_size = 10)\n",
+    "loader = Loader(batcher)\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    for batch in loader:\n",
+    "        print(f\"Epoch = {epoch}\", f\" Batch = {batch}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "860142c7-a39b-441b-b2b2-1cbd81a7a0f3",
+   "metadata": {},
+   "source": [
+    "#### SamplerWrapper"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "18134e7d-d0e8-4096-a674-23e715fa16a3",
+   "metadata": {},
+   "source": [
+    "As mentioned earlier, we can use `torch.data.utils` samplers using `MapStyleWrapper`.\n",
+    "Alternatively, we can employ the `SamplerWrapper`, which effectively converts a `Sampler` into a `BaseNode`, which can be further used for generating batches. Moreover, `SamplerWrapper` also tracks the number of epochs."
```

Comment:
```suggestion
    "Alternatively, we can employ the `SamplerWrapper`, which converts a `Sampler` into a `BaseNode`. `SamplerWrapper` differs from `IterableWrapper` because it will track the number of epochs, and call the sampler's `set_epoch` method if it is implemented."
```